pkg/datapath/iptables/iptables.go

```diff
@@ -125,6 +125,25 @@ func getFeedRule(name, args string) []string {
 	return append(argsList, ruleTail...)
 }
 
+// skipPodToPodConntrack returns true if it's possible to install iptables
+// `-j NOTRACK` rules to skip tracking pod to pod traffic.
+//
+// These rules will be installed only when Cilium is running in strict KPR mode
+// (as otherwise conntrack would be required for NAT operations) and in direct
+// routing mode (as in tunneling mode the encapsulated traffic is already
+// skipping netfilter conntrack).
+//
+// Moreover, these rules can only be installed for IPv4 traffic, as the native
+// routing CIDR, used to select all pod to pod traffic, can only be an IPv4 CIDR
+// at the moment.
+func skipPodToPodConntrack(ipv6 bool) bool {
+	return !ipv6 && option.Config.EnableIPv4 &&
+		option.Config.InstallNoConntrackIptRules &&
+		option.Config.Tunnel == option.TunnelDisabled &&
+		option.Config.IPv4NativeRoutingCIDR() != nil &&
+		option.Config.KubeProxyReplacementIsRunningStrict()
```

Comment:
We may also need to disable if CNI chaining is enabled. I'm not too sure how that would play with e.g. Azure CNI and AWS-CNI.

Reply:
OK, I'm also convinced that we should disable it (at least for now) when CNI chaining is enabled: looking at the docs there are [quite a few different supported configurations](https://docs.cilium.io/en/v1.9/gettingstarted/cni-chaining/) and each of them would require some more thinking to determine if we can relax this constraint.

I followed the same approach I used to the managed Kubernetes environment and disable it in the helm config-map template: https://github.com/cilium/cilium/blob/99ff3b55e2f68277236b703866a026e6a2f17824/install/kubernetes/cilium/templates/cilium-configmap.yaml#L440